File: week4/rlawjddbs/PizzaStore.java
package week4;

public abstract class PizzaStore {
	
	public Pizza orderPizza(String type) {
		Pizza pizza;
		
		pizza = createPizza(type);
		
		pizza.prepare();
		pizza.bake();
		pizza.cut();
		pizza.box();
		
		return pizza;
	} // orderPizza
	
	protected abstract Pizza createPizza(String type);
	
} // class

File: week4/ch.Choe/Week4/Week4/Store/PizzaStore.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Week4.IngredientFactory;

namespace Week4
{
    public abstract class PizzaStore
    {
        public PizzaIngredientFactory IngredientFactory;
        public Pizza OrderPizza(string type, StaticSimpleCuttingFactory.SliceType slicetype)
        {
            Pizza pizza;
            pizza = CreatePizza(type, slicetype);
            pizza.Prepare();
            pizza.Bake();
            pizza.Cut();
            pizza.Box();

            return pizza;   
        }

        protected abstract Pizza CreatePizza(string type, StaticSimpleCuttingFactory.SliceType slicetype);
        
    }
}
